src/stream.rs

```diff
@@ -105,7 +105,19 @@ impl<T: AsyncRead + AsyncWrite + Connection + Unpin> Connection for MaybeHttpsSt
     fn connected(&self) -> Connected {
         match self {
             MaybeHttpsStream::Http(s) => s.connected(),
-            MaybeHttpsStream::Https(s) => s.get_ref().get_ref().get_ref().connected(),
+            MaybeHttpsStream::Https(s) => {
+                let tls = s.get_ref();
+                if tls
+                    .negotiated_alpn()
+                    .ok()
```

Comment:
The error is somewhat silently ignored here, probably not great.

Reply:
Ignoring it here seems fine.